llvm-external-projects/iree-dialects/include/iree-dialects/Dialect/LinalgExt/IR/LinalgExtOps.td

```diff
@@ -890,6 +890,94 @@ let builders = [
   }];
 }
 
+//===----------------------------------------------------------------------===//
+// Winograd ops
+//===----------------------------------------------------------------------===//
+
+def IREELinalgExt_WinogradInputTransformOp : IREELinalgExt_Op<"winograd.input_transform",
+    [DeclareOpInterfaceMethods<ReifyRankedShapedTypeOpInterface>,
+     DeclareOpInterfaceMethods<TilingInterface,
+      ["getIterationDomain",
+       "getLoopIteratorTypes",
+       "getResultTilePosition",
+       "getTiledImplementation"]>]> {
+  let summary = "Winograd Input Transform operator";
+  let description = [{
+    This operator is the first step in converting a convolution to
+    its Winograd equivalent. Given a tile of an input image (I),
+    this operator computes matmul(tranpose(B), matmul(I, B)).
+    The input tile is assumed to be square with each side of size m + r - 1,
+    where the convolutional kernel is m x m and the output tile size is r x r.
+    B is a constant 2-d square matrix of the same shape as the input tile I.
+    The input to the operator is an image of shape (N, H, W, C) and the
+    output is an operator of shape (m + r - 1, m + r - 1, N, H', W', C)
+    where H' = ceil((H - m + 1)/r) and W' = ceil((W - m + 1)/r). The result
+    of this operator is first collapsed and then fed to a batch matmul op.
+  }];
+
+  let arguments = (ins Variadic<AnyShaped>:$inputs,
+                       Variadic<AnyShaped>:$outputs,
+                       I64Attr:$output_tile_size,
+                       I64Attr:$kernel_size,
+                       StrAttr:$tensor_format
+  );
+
+  let builders = [
+    OpBuilder<(ins "ValueRange":$inputs, "ValueRange":$outputs,
+      CArg<"int64_t", "8">:$output_tile_size, CArg<"int64_t", "3">:$kernel_size,
+      CArg<"StringRef", "\"nhwc\"">:$tensor_format)>
+  ];
+
+  let results = (outs Variadic<AnyRankedTensor>:$result);
+  let hasFolder = 1;
+  let assemblyFormat = [{
+    attr-dict
+    `output_tile_size` `(` $output_tile_size `)`
+    `kernel_size` `(` $kernel_size `)`
+    `tensor_format` `(` $tensor_format `)`
```

Comment:
Instead of `tensor_format`, I am guessing you only need to know which are the image dimensions... So you can just replace this with same `image_dimensions` that is an `IntegerArrayAttr` that returns the image dimensions.

Reply:
yes I just need to know which dims to expose to the tiling interface and which not to. Sure will change.